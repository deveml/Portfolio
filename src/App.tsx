import { StyledEngineProvider, ThemeProvider, Typography } from "@mui/material";
import { lazy, Suspense } from "react";
import { FallBack } from "./components/Fallback";
import { appTheme } from "./theme/theme";

/**
 * @description Lazy loading fonts so that the render will not be blocked
 */
const LazyFontLoader = lazy(() => import("./theme/FontLoader"));

function App() {
  return (
    <Suspense fallback={<FallBack />}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={appTheme}>
          <Typography>Sample app</Typography>
          <LazyFontLoader />
        </ThemeProvider>
      </StyledEngineProvider>
    </Suspense>
  );
}

export default App;
