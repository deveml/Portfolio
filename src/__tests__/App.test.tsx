import * as renderer from "react-test-renderer";
import App from "../App";

describe("App wrapper component snapshot test", () => {
  it("Should render the component and it should match the stored snapshot", () => {
    const appComponent = renderer.create(<App />);
    const parsedComponent = appComponent.toJSON();
    expect(parsedComponent).toMatchSnapshot();
  });
});

export {};
