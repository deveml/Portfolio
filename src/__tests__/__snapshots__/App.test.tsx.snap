// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App wrapper component snapshot test Should render the component and it should match the stored snapshot 1`] = `
[
  "Uh-oh! We just ran into an error",
]
`;
